src/NzbDrone.Core/Indexers/Definitions/FileList/FileListRequestGenerator.cs

```diff
@@ -13,108 +15,140 @@ public class FileListRequestGenerator : IIndexerRequestGenerator
         public Func<IDictionary<string, string>> GetCookies { get; set; }
         public Action<IDictionary<string, string>, DateTime?> CookiesUpdater { get; set; }
 
-        public virtual IndexerPageableRequestChain GetSearchRequests(MovieSearchCriteria searchCriteria)
+        public IndexerPageableRequestChain GetSearchRequests(TvSearchCriteria searchCriteria)
         {
             var pageableRequests = new IndexerPageableRequestChain();
+            var parameters = GetDefaultParameters();
 
-            if (searchCriteria.ImdbId.IsNotNullOrWhiteSpace())
-            {
-                pageableRequests.Add(GetRequest("search-torrents", searchCriteria.Categories, string.Format("&type=imdb&query={0}", searchCriteria.FullImdbId)));
-            }
-            else if (searchCriteria.SearchTerm.IsNotNullOrWhiteSpace())
+            if (searchCriteria.ImdbId.IsNotNullOrWhiteSpace() || searchCriteria.SearchTerm.IsNotNullOrWhiteSpace())
             {
-                var titleYearSearchQuery = string.Format("{0}", searchCriteria.SanitizedSearchTerm);
-                pageableRequests.Add(GetRequest("search-torrents", searchCriteria.Categories, string.Format("&type=name&query={0}", titleYearSearchQuery.Trim())));
-            }
-            else
-            {
-                pageableRequests.Add(GetRequest("latest-torrents", searchCriteria.Categories, ""));
+                parameters.Add("action", "search-torrents");
+
+                if (searchCriteria.ImdbId.IsNotNullOrWhiteSpace())
+                {
+                    parameters.Add("type", "imdb");
+                    parameters.Add("query", searchCriteria.FullImdbId);
+                }
+                else if (searchCriteria.SearchTerm.IsNotNullOrWhiteSpace())
+                {
+                    parameters.Add("type", "name");
```

Comment:
Should we separate this into a separate clause. What if user searches with `Imdb` and uses `q` for season or episode string?

Reply:
Ah yea, forgot that's a type switch. All good